### PR TITLE
Update the token on session regeneration

### PR DESCRIPTION
**Pull request by rullzer**
More abstraced form of https://github.com/nextcloud/server/pull/9756

### CODE REVIEW

File: lib/public/ISession.php [hunk 1]
```diff
@@ -96,10 +96,11 @@ public function close();
 	 * Wrapper around session_regenerate_id
 	 *
 	 * @param bool $deleteOldSession Whether to delete the old associated session file or not.
+	 * @param bool $updateToken Wheater to update the associated auth token
 	 * @return void
 	 * @since 9.0.0
```
Comment (MorrisJobke) [inline lib/public/ISession.php]: Document that `$updateToken` was added in 14.0.0

**Review by MorrisJobke — APPROVED** [review]
Code makes sense beside my little nitpick.

File: lib/public/ISession.php (hunk 1 above)
Comment (rullzer) [reply lib/public/ISession.php]: Done

**Review by blizzz — APPROVED** [review]

**Comment by MorrisJobke** [comment]
@rullzer What about the backport here?

**Pull request merged**